data/block/unmarshal.go

```diff
@@ -0,0 +1,25 @@
+package block
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+)
+
+// GetHeaderFromBytes will unmarshal the header bytes based on the header type
+func GetHeaderFromBytes(marshaller marshal.Marshalizer, creator EmptyBlockCreator, headerBytes []byte) (data.HeaderHandler, error) {
```

Comment:
yes, it can be done, of course. Although, as far as I know, all the processing usages of this function already contain the marshaller, so it is easier to provide that. I suggest just add a component that call this (still exported) function and later we can decide what to use.